src/shared.rs

```diff
@@ -184,6 +184,14 @@ macro_rules! glib_shared_wrapper {
             }
         }
 
+        #[doc(hidden)]
+        impl $crate::translate::FromGlibPtrBorrow<*const $ffi_name> for $name {
+            #[inline]
+            unsafe fn from_glib_borrow(ptr: *const $ffi_name) -> Self {
+                $name($crate::translate::from_glib_borrow(ptr as *mut $ffi_name))
```

Comment:
I believe you can also do
```
$crate::translate::from_glib_borrow(ptr as *mut $ffi_name)
```
here and in the other places. Without the `$name(...)` part.

Reply:
Isn't it what I did?

Reply:
Oh nevermind.